WordPress/gutenberg
# Initial prototype of simplified layout panel

## What?


Starts moving towards the latest designs in #42385.

New layout panel controls vary depending on whether layout switching is allowed (such as for the Group block with its Row, Stack, and experimental Grid variations) or what the layout type is. E.g.:

Flex block:
<img width="276" alt="Screenshot 2023-05-05 at 1 56 41 pm" src="https://user-images.githubusercontent.com/8096000/236375133-324c4404-c165-4006-9f65-c7703a36260a.png">

Flow/constrained block:
<img width="278" alt="Screenshot 2023-05-05 at 1 59 13 pm" src="https://user-images.githubusercontent.com/8096000/236375149-eb760aeb-4573-4e5d-af9a-d00d645b3c6a.png">

Switchable block:
<img width="279" alt="Screenshot 2023-05-05 at 1 57 08 pm" src="https://user-images.githubusercontent.com/8096000/236375183-0b4aa8b6-7b65-4c1f-9a8a-a3d3a40da9e6.png">

As it stands, these changes don't affect the layout API in any way, except for a small difference in behaviour for flex layouts: with vertical orientation, flex can now be switched to flow or constrained layout by toggling the content width control. This still causes a little weirdness in Buttons and Navigation blocks due to their custom styles and markup structures, but that can be fixed by updating the styles in those blocks. 

Another side-effect of this PR is that the `allowSwitching` option now works very well, whereas on trunk it still looks somewhat broken.

## Testing Instructions





1. Add blocks with layout to a post or template, such as Group, Buttons, Social Icons, Navigation, Columns, Post Content.
2. Try all the controls and check that everything behaves as expected.


### Testing Instructions for Keyboard


## Screenshots or screencast
